### ISSUE TITLE
Emulator does not return HTTP status code for cloud functions

### ISSUE THREAD

**Issue by c-2002** [firebase/firebase-tools#1257]
### [REQUIRED] Environment info

**firebase-tools:**

```
> firebase --version
6.9.0
```

**Platform:**

- OSX 10.14.4
- node.js: v10.15.3
- npm: 6.9.0

### [REQUIRED] Test case

Create a new Firebase project using firebase tools and implement a simple helloWord cloud function. Call the cloud function and check the HTTP status code.

### [REQUIRED] Steps to reproduce

Create a new project in a shell:

```
mkdir test
cd test
firebase init
```

Choose the following options:

- Firebase CLI features: Functions
- Existing project or create a new one (it does not matter)
- Language: JavaScript
- ESLint: No
- Install npm: Yes

Open functions/index.js with Visual Studio Code or any other text editor and create the following node.js code:

```
exports.helloWorld = functions.https.onRequest((request, response) => {
  response.status(500);
  response.send("Hello from Firebase!");
});
```

After that, install the firebase-functions-test npm with the command:

```
npm install --save-dev firebase-functions-test
```

In the package.json set node.js version 10.

```
{
  "name": "functions",
  "description": "Cloud Functions for Firebase",
  "scripts": {
    "serve": "firebase serve --only functions",
    "shell": "firebase functions:shell",
    "start": "npm run shell",
    "deploy": "firebase deploy --only functions",
    "logs": "firebase functions:log"
  },
  "engines": {
    "node": "10"
  },
  "dependencies": {
    "firebase-admin": "~7.0.0",
    "firebase-functions": "^2.3.0"
  },
  "private": true
}
```

Install all npm files using the commands

```
cd functions
npm install
```

and start the Firebase emulator in a terminal

```
firebase emulators:start
```

The output will like the one below:

```
firebase emulators:start
i  Starting emulators: ["functions"]
✔  functions: Using node@10 from host.
i  functions: Watching "/Users/dev-user/dev/test/functions" for Cloud Functions...
i  functions: HTTP trigger initialized at http://localhost:5001/my-sample-app/us-central1/helloWorld
```

Next, call the cloud function that is provided by the emulator using curl to check the http status code.

```
curl http://localhost:5001/my-sample-app/us-central1/helloWorld -v

```

The result will be:

```
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 5001 (#0)
> GET /my-sample-app/us-central1/helloWorld HTTP/1.1
> Host: localhost:5001
> User-Agent: curl/7.54.0
> Accept: */*
>
< HTTP/1.1 200 OK
< X-Powered-By: Express
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Headers: Origin, X-Requested-With, Content-Type, Accept
< Date: Wed, 08 May 2019 07:21:45 GMT
< Connection: keep-alive
< Transfer-Encoding: chunked
<
* Connection #0 to host localhost left intact
Hello from Firebase!
```

Stop the emulator and start

```
firebase serve --only functions
```

Run the curl command from above again. The result will be the same.
In both cases the local Firebase return a HTTP 200 code.

Deploy the cloud function to the Firebase servers:

```
firebase deploy --only functions
```

... and run the curl command to the Firebase host URL:

```
curl https://us-central1-my-sample-app.cloudfunctions.net/helloWorld -v
```

The result will be:

```
*   Trying 172.217.17.238...
* TCP_NODELAY set
* Connected to us-central1-my-sample-app.cloudfunctions.net (172.217.17.238) port 443 (#0)

(.... lines removed to shorten output ...)

> GET /helloWorld HTTP/2
> Host: us-central1-my-sample-app.cloudfunctions.net
> User-Agent: curl/7.54.0
> Accept: */*
>
* Connection state changed (MAX_CONCURRENT_STREAMS updated)!
< HTTP/2 500
< content-type: text/html; charset=utf-8
< etag: W/"14-z3iZXchEt5DVWZKsMncy8Wl4KSQ"
< function-execution-id: cp2997brzyn2
< x-powered-by: Express
< x-cloud-trace-context: 664d3dfc1d7bbda0a1107adeee933847;o=1
< date: Wed, 08 May 2019 07:46:34 GMT
< server: Google Frontend
< content-length: 20
< alt-svc: quic=":443"; ma=2592000; v="46,44,43,39"
<
* Connection #0 to host us-central1-my-sample-app.cloudfunctions.net left intact
Hello from Firebase!
```

The hosted function returns a HTTP 500 code.

### [REQUIRED] Expected behavior

The local execution always return a HTTP 200 code. The node.js implementation sets the code to HTTP 500. I expect the the local execution als returns a HTTP 500 code.

### [REQUIRED] Actual behavior

The Firebase emulator call always returns a HTTP 200 code. It does not consider the status set in the node.js code.

**Comment by samtstern** [comment]
Thanks for reporting!  This was fixed by #1246 and will be released soon.  Please downgrade to `6.8.0` for now.

**Comment by samtstern** [comment]
The fix for this issue has been released in `6.9.1`.
